### ISSUE TITLE
Cannot open document

### ISSUE THREAD

**Issue by blevinscm** [microsoft/vscode-cosmosdb#1732]
Does this occur consistently? 
Repro steps:

1. Connect to Cosmos Account
2. Expand DB in the task pane
3. Select document
4. Nothing happens and error throws

Action: cosmosDB.openDocument
Error type: EntryNotFound
Error Message: EntryNotFound (FileSystemError): azureDatabases:/Johnson_dc8c3127-4318-4830-9a18-8d9054c66687-cosmos-document.json?id=/subscriptions/1e555350-e8a6-4258-b56f-3005e3c59e5b/resourceGroups/blev/providers/Microsoft.DocumentDB/databaseAccounts/blevtest/AzureSampleFamilyDatabase/FamilyContainer/$Documents/iWZFAKqnx+4DAAAAAAAAAA==

Version: 0.15.0
OS: win32
OS Release: 10.0.20180
Product: Visual Studio Code - Insiders
Product Version: 1.51.0-insider
Language: en

<details>
<summary>Call Stack</summary>

```
file:/workbench.desktop.main.js:6654:715
```

</details>

**Comment by neelip** [comment]
Are you see this issue with only this specific document or with all documents?

**Comment by blevinscm** [comment]
It happens only with that document as far as I can tell.

**Comment by Alexander-Vasilenko** [comment]
I ran into the same issue while I was working on Learn/Work with NoSQL data in Azure Cosmos DB/Build a .NET Core app for Azure Cosmos DB in Visual Studio Code. I used the code from the 4th unit and I cannot open both created documents. I also added 3rd document to the db through the portal and tried to open it but result was the same.

Does this occur consistently? 
Yes.
Repro steps:

1. Choose Concierge Subscription
2. Expand Cosmos DB account pane
3. Expand database pane
4. Expand collection pane
5. Expand document pane
6. Click on document 1 or 2

Actual Result:
Document doesn't open and user get an error with the button to report it.

Action: cosmosDB.openDocument
Error type: EntryNotFound
Error Message: EntryNotFound (FileSystemError): azureDatabases:/1-cosmos-document.json?id=/subscriptions/be8b0530-081c-4940-9f0c-f15fa2e65980/resourceGroups/learn-1dc9be3f-f86b-4af8-a624-e5b5aedaff1d/providers/Microsoft.DocumentDB/databaseAccounts/asvas-cosmos-db-sql/Users/WebCustomers/$Documents/+wxGAIvUUioBAAAAAAAAAA==

Version: 0.15.1
OS: win32
OS Release: 10.0.18363
Product: Visual Studio Code
Product Version: 1.51.1
Language: en

<details>
<summary>Call Stack</summary>

```
file:/workbench.desktop.main.js:6687:388
```

</details>
Action: cosmosDB.openDocument
Error type: EntryNotFound
Error Message: EntryNotFound (FileSystemError): azureDatabases:/2-cosmos-document.json?id=/subscriptions/be8b0530-081c-4940-9f0c-f15fa2e65980/resourceGroups/learn-1dc9be3f-f86b-4af8-a624-e5b5aedaff1d/providers/Microsoft.DocumentDB/databaseAccounts/asvas-cosmos-db-sql/Users/WebCustomers/$Documents/+wxGAIvUUioCAAAAAAAAAA==

Version: 0.15.1
OS: win32
OS Release: 10.0.18363
Product: Visual Studio Code
Product Version: 1.51.1
Language: en

<details>
<summary>Call Stack</summary>

```
file:/workbench.desktop.main.js:6687:388
```

</details>

**Comment by joelybahh** [comment]
I'm encountering the same issue. All documents inside one of my collections fail to open in VSCode. All my other collections documents open fine.  

`azureDatabases:/view_d17410a46b884069a194b27dbad54c89-cosmos-document.json?id%3D%2Fsubscriptions%2F22abb5cb-2b1f-4204-a1a5-2f0b611f04f1%2FresourceGroups%2FCloud%2Fproviders%2FMicrosoft.DocumentDB%2FdatabaseAccounts%2Finspace-cloud%2Fapp%2Fviews%2F%24Documents%2Fl59AAPV%2BQe5JAAAAAAAAAA%3D%3D. Detail: Unable to read file 'azureDatabases:/view_d17410a46b884069a194b27dbad54c89-cosmos-document.json?id=/subscriptions/22abb5cb-2b1f-4204-a1a5-2f0b611f04f1/resourceGroups/Cloud/providers/Microsoft.DocumentDB/databaseAccounts/inspace-cloud/app/views/$Documents/l59AAPV+Qe5JAAAAAAAAAA==' (Error: Unable to resolve non-existing file 'azureDatabases:/view_d17410a46b884069a194b27dbad54c89-cosmos-document.json?id=/subscriptions/22abb5cb-2b1f-4204-a1a5-2f0b611f04f1/resourceGroups/Cloud/providers/Microsoft.DocumentDB/databaseAccounts/inspace-cloud/app/views/$Documents/l59AAPV+Qe5JAAAAAAAAAA==')`

**Comment by neelip** [comment]
Hi @joelybahh, it looks like your error is different from the other errors presented in this issue. Let us know if you are still getting the "Unable to read file..." error, we can create a new issue for it.

**Comment by joelybahh** [comment]
> Hi @joelybahh, it looks like your error is different from the other errors presented in this issue. Let us know if you are still getting the "Unable to read file..." error, we can create a new issue for it.

Hey @neelip , Yes I am still getting this issue, was there any way I can get a more detailed error log for you all, to help debug 
further?

**Comment by joelybahh** [comment]
@neelip was there any update on the status of this error? It is still occurring on VS Code, its the only cosmos container that does it, all other containers work fine, just all files within that one container won't open.

`cannot open azureDatabases:/view_466ffba7e78c4ba68f2f9437247c8737-cosmos-document.json?id%3D%2Fsubscriptions%2F22abb5cb-2b1f-4204-a1a5-2f0b611f04f1%2FresourceGroups%2FCloud%2Fproviders%2FMicrosoft.DocumentDB%2FdatabaseAccounts%2Finspace-cloud%2Fprovisionary%2Fviews%2F%24Documents%2FqlAlAJ%2BlkPs9AAAAAAAAAA%3D%3D. Detail: Unable to read file 'azureDatabases:/view_466ffba7e78c4ba68f2f9437247c8737-cosmos-document.json?id=/subscriptions/22abb5cb-2b1f-4204-a1a5-2f0b611f04f1/resourceGroups/Cloud/providers/Microsoft.DocumentDB/databaseAccounts/inspace-cloud/provisionary/views/$Documents/qlAlAJ+lkPs9AAAAAAAAAA==' (Error: Unable to resolve non-existing file 'azureDatabases:/view_466ffba7e78c4ba68f2f9437247c8737-cosmos-document.json?id=/subscriptions/22abb5cb-2b1f-4204-a1a5-2f0b611f04f1/resourceGroups/Cloud/providers/Microsoft.DocumentDB/databaseAccounts/inspace-cloud/provisionary/views/$Documents/qlAlAJ+lkPs9AAAAAAAAAA==')`

![image](https://user-images.githubusercontent.com/6394680/124840139-a5ed1580-dfcd-11eb-96a9-656a6f9878d1.png)
The above image is just demonstrating what I mean by the whole container. every file in `views` will throw that error, but every other container (Marked with green tick), all the documents within those open fine

**Comment by neelip** [comment]
Hi @joelybahh, sorry for the delay. I think you error might be something to do with partition key/value for the collection, either you have a key but no value or you don't have a key at all. Please try to check the key and the value are correctly set.

**Comment by neelip** [comment]
Most of the EntryNotFound errors are happening due a problem with how partition key/value are set. Best option would be to throw a more readable error or update the "Enter partition key[or value]..." prompt.

**Comment by mikebourbeauart** [comment]
> Most of the EntryNotFound errors are happening due a problem with how partition key/value are set. Best option would be to throw a more readable error or update the "Enter partition key[or value]..." prompt.

I'm having this same issue. I'm not really sure how to proceed. The tutorial says to not enter a key and just press enter. If we're supposed to put a key, what should it be?

I'm on this part of the Cosmos DB/Node.js tutorial btw: https://docs.microsoft.com/en-us/learn/modules/build-node-cosmos-app-vscode/3-exercise-create-cosmosdb-instance

**Comment by GaryParr** [comment]
I'm getting the same "unable to resolve nonexistent file" for every document in every collection within a database.

**Comment by Zerg00s** [comment]
I'm getting the same error:

![image](https://user-images.githubusercontent.com/2797648/163474109-4c2dd27b-a3d2-498c-a912-1e4f952b1497.png)

**Comment by Dillon-Subvrsive** [comment]
Not good news for those who have data they need to keep, but I was able to resolve this by deleting my newly created Collection and recreating.

I'm not sure if perhaps I did a bad Bicep/ARM deployment involving the PartitionKey perhaps, but I'm not going to spend the time to attempt to recreate the problem.

**Comment by marked23** [comment]
This is happening to me when targeting Azure Cosmos DB Emulator.  All the setup of the database and collections was done through the VSCode UI.  I have my partition key set to "id" (which is a guid as a string), if that matters.
![image](https://github.com/microsoft/vscode-cosmosdb/assets/732793/44bf4fbf-889e-4513-9f22-21b0e2a33030)

However, if I open the Data Explorer UI that comes with the emulator, I can view the documents just fine.
![image](https://github.com/microsoft/vscode-cosmosdb/assets/732793/f6660299-9bce-4370-89e2-313865f84411)

**Comment by JasonYeMSFT** [comment]
I merged a fix to mitigate this issue. The root cause was we were using _rid as the "id" parameter in the document's uri. In uri query string standard, plus sign '+' is decoded to whitespace ' ' for backward compatibility. In the extension, we cache the document with the _rid as is. However, after putting the _rid with the '+' to the query string and parse them back, the '+' is replaced by a whitespace and we can no longer match the correct document we have cached. The fix is to use a more appropriate value in the query parameter.

Although I merged a PR switching to use the id + partition combo, I realized it may still have this problem if your id has '+' or even '%2B' (the url encoded form of '+'). I expect it to be less likely though. To further mitigate this problem, I am going to replace the '+' sign to something definitely illegal in the original resource id. I am not going to deal with the '%2B's in the resource id.
